### DISCUSSION TITLE
Something has gone seriously wrong: import_mok_state() failed: Not Found

### DISCUSSION THREAD

**Discussion by sean-skt** [linux-surface/linux-surface#1256] [Q&A] [answered]
I have a Microsoft Surface Book 2 and have the following issue:

I initially installed Pop!_OS 22.04 alongside Windows 11. Secure Boot had to be off to do this.

I then proceeded to install the linux-surface kernel into Pop!_OS. Then I rebooted. After that I followed the instructions to install linux-surface-secureboot-mok.

On reboot, nothing happened and instead the Surface Book 2 just rebooted back into Pop!_OS.

On subsequents attempts to boot from the Pop!OS 22.04 USB installer I got the following error:

Failed to open \EFI\BOOT\mmx64.efi - Not Found
Failed to load image : Not Found
Failed to start MokManager: Not Found
Something has gone seriously wrong: import_mok_state() failed: Not Found

I also found that now other distro's USB installers also no longer boot. I realize now that I am having these issues because I dodn't follow the instructions properly. Is there a way that I can reset things before I had these issues with Pop OS? Thank you very much.

**Comment by sean-skt** [comment]
Please guys, I really need a hand here. Admittedly I have gotten myself into a rotten mess here because of the following:

1. I shouldn't have tried this on Pop!_OS
2. I shouldn't have disabled secure boot BEFORE installing the "linux-surface" kernel and then rebooting again before issuing the following command: "sudo apt install linux-surface-secureboot-mok".

The problem is I cannot boot any Linux distro (or any of the USB installers) with Secure Boot on. It appears that my issues stem from something that has changed in the UEFI NVRAM.

**Reply by StollD** [reply]
Does Pop OS even support secureboot? AFAIK they use systemd-boot, which Microsoft doesn't sign.

Can you still boot your Pop OS install? If yes, disable secureboot, boot into Pop OS, and then run `sudo mokutil --revoke-import`

**Comment by sean-skt** [comment]
Thanks for you reply.
No Pop!_OS doesn't support Secure Boot as I mentioned in my 1st post. This is why I have run into these problems.

I have ditched Pop!_OS and am using Debian 12. But I cannot use the default kernel in Debian with Secure Boot.
To even get Debian 12 to boot now, I have turn off secure boot and rename grubx64.efi to mmx64.efi in the EFI partition.

If I try to boot Debian 12 through Grub, with secure boot on I get the following error message:

error: shim_lock protocol not found.
error: you need to load the kernel first.

All my problems began in Pop!_OS 22.04 when using "sudo apt install linux-surface-secureboot-mok" AFTER: (1)  I had disabled secure boot, (2) installed the linux-surface kernal and (3) rebooted.

Someone mentioned that perhaps the installation set a few variables in the UEFI variable store and it's trying to initialize the MOKmanager. I think I may need to undo that and don't know how.

sudo mokutil --revoke-import has not helped.

**Comment by sean-skt** [comment, marked as answer]
I have a working bootable UEFI shell on a USB drive. It does require secure boot to be off.

Are there any commands that can be used to undo whatever it was that "sudo apt install linux-surface-secureboot-mok" did in Pop OS (with the linux-surface kernel already installed and secure boot off)?

EDIT: OK, after very many days trying to sort this out I figured it out.
Had to boot into a UEFI shell and delete the following variables:

dmpstore -d -all MokAuth
dmpstore -d -all MokNew
dmpstore -d -all MokSB

All is back to the way it was before I stuffed things up in Pop!_OS.

**Reply by cgiAlexis** [reply]
For future searchers, I combined sean-skt's guide with [this Baledung guide](https://www.baeldung.com/linux/efi-shell-bootable-flash) in order to create a bootable EFI shell to get to this stage and fix the problem.

**Reply by RealDekkia** [reply]
Alternatively to the guide cgiAlexis linked, you can also download https://github.com/tianocore/edk2/raw/edk2-stable201903/ShellBinPkg/UefiShell/X64/Shell.efi (from the linked guide), name it BOOTX64.efi and put it into efi/boot/ on the thumb drive you already have prepared to install Linux with. Rename the existing bootx64.efi before doing that, so you don't have to re-create the bootable stick once you're done.

**Reply by royjimmy286-code** [reply]
Can you plz explain it to me step by step im detail am in a lot of trouble plz anyone
I’d appreciate it🙏🙏🙏

**Reply by royjimmy286-code** [reply]
> Alternatively to the guide cgiAlexis linked, you can also download https://github.com/tianocore/edk2/raw/edk2-stable201903/ShellBinPkg/UefiShell/X64/Shell.efi (from the linked guide), name it BOOTX64.efi and put it into efi/boot/ on the thumb drive you already have prepared to install Linux with. Rename the existing bootx64.efi before doing that, so you don't have to re-create the bootable stick once you're done.

Rename to what exactly why not delete old one

**Reply by RealDekkia** [reply]
> > Alternatively to the guide cgiAlexis linked, you can also download https://github.com/tianocore/edk2/raw/edk2-stable201903/ShellBinPkg/UefiShell/X64/Shell.efi (from the linked guide), name it BOOTX64.efi and put it into efi/boot/ on the thumb drive you already have prepared to install Linux with. Rename the existing bootx64.efi before doing that, so you don't have to re-create the bootable stick once you're done.
> 
> Rename to what exactly why not delete old one

If you delete the old one you have to re-create the installer USB-stick after you ran the commands.
If you rename it to something else, you just have to change the name back afterwards.

**Comment by royjimmy286-code** [comment]
Love you man,thanks
